prov/efa/src/efa_cq.h

```diff
@@ -148,11 +152,31 @@ static inline int efa_cq_ibv_cq_ex_open(struct fi_cq_attr *attr,
 		efadv_cq_init_attr.wc_flags |= EFADV_WC_EX_WITH_IS_UNSOLICITED;
 #endif
 
+#if HAVE_CAPS_CQ_WITH_EXT_MEM_DMABUF
+	if (efa_device_support_cq_with_ext_mem_dmabuf() &&
```

Comment:
Shouldn't we return some error when external memory was provided but not supported? (check requested vs. supported flags)

Reply:
Also it's very strange that the dmabuf info is passed as an arg to the function but the decision whether to use is comes from another place.